packages/@aws-cdk/aws-kinesisfirehose-destinations/lib/redshift-cluster.ts

```diff
@@ -0,0 +1,276 @@
+   *
+   * Used to create the user that Firehose assumes and the table that data is inserted into.
+   *
+   * @default - the master secret is taken from the cluster.
+   */
+  readonly masterSecret?: secretsmanager.ISecret;
+
+  /**
+   * Parameters given to the COPY command that is used to move data from S3 to Redshift.
+   *
+   * @see https://docs.aws.amazon.com/firehose/latest/APIReference/API_CopyCommand.html
+   *
+   * @default '' - no extra parameters are provided to the Redshift COPY command
+   */
+  readonly copyOptions?: string;
+
+  /**
+   * The length of time during which Firehose retries delivery after a failure.
+   *
+   * @default Duration.hours(1)
+   */
+  readonly retryTimeout?: cdk.Duration;
+
+  /**
+   * The intermediate bucket where Firehose will stage your data before COPYing it to the Redshift cluster.
+   *
+   * @default - a bucket will be created for you.
+   */
+  readonly intermediateBucket?: s3.IBucket;
+
+  /**
+   * The role that is attached to the Redshift cluster and will have permissions to access the intermediate bucket.
+   *
+   * If a role is provided, it must be already attached to the cluster, to avoid the 10 role per cluster limit.
+   *
+   * @default - a role will be created for you.
+   */
+  readonly bucketAccessRole?: iam.IRole;
+}
+
+/**
+ * Redshift delivery stream destination.
+ */
+export class RedshiftCluster implements firehose.IDestination {
+  /**
+   * The secret Firehose will use to access the Redshift cluster.
+   */
+  public secret?: secretsmanager.ISecret;
+
+  private masterSecret: secretsmanager.ISecret;
+
+  constructor(private readonly cluster: redshift.ICluster, private readonly props: RedshiftDestinationProps) {
+    if (!cluster.publiclyAccessible) {
+      throw new Error('Redshift cluster used as Firehose destination must be publicly accessible');
+    }
+    if (!cluster.subnetGroup?.selectedSubnets?.hasPublic) {
+      throw new Error('Redshift cluster used as Firehose destination must be located in a public subnet');
+    }
+    const masterSecret = props.masterSecret ?? (cluster instanceof redshift.Cluster ? cluster.secret : undefined);
+    if (!masterSecret) {
+      throw new Error('Master secret must be provided or Redshift cluster must generate a master secret');
+    }
+    this.masterSecret = masterSecret;
+    if (props.retryTimeout && props.retryTimeout.toSeconds() > cdk.Duration.hours(2).toSeconds()) {
+      throw new Error('Retry timeout must be less that 2 hours');
+    }
+    if (props.compression === Compression.SNAPPY || props.compression === Compression.HADOOP_SNAPPY || props.compression === Compression.ZIP) {
+      throw new Error('Compression must not be HADOOP_SNAPPY, SNAPPY, or ZIP');
+    }
+    if (props.s3Backup?.mode === BackupMode.FAILED) {
+      throw new Error('Redshift delivery stream destination only supports ALL and DISABLED BackupMode, given FAILED');
```

Comment:
DISABLED backup mode doesn't exist anymore in the CDK API. 
```suggestion
      throw new Error('Redshift delivery stream destination does not support FAILED BackupMode.');
```